src/components/auth/Register.jsx

```diff
@@ -9,6 +14,48 @@ export const Register = () => {
       formState: { errors }
    } = useForm()
 
+   const dispatch = useDispatch()
+   const { message, status } = useSelector((state) => state.user)
+   const [alertMessage, setAlertMessage] = useState({
+      message: '',
+      backgroundColor: ''
+   }) 
+   const isAuthenticating = useMemo(() => status === 'checking', [status])
+
+
+   const handleAlertMessage = (alertMessage) => {
+      switch (alertMessage) {
+         case 'Firebase: Error (auth/email-already-in-use).': {
+            setAlertMessage({
+               message: 'Email already in use.',
+               backgroundColor: 'bg-red-500'
+            })
+            break
+         }
+         case 'Register successful': {
+            setAlertMessage({
+               message: 'Register successful!',
+               backgroundColor: 'bg-green-500'
+            })
+            break
+         }
+         case 'Checking...': {
+            setAlertMessage({
+               message: 'Checking...',
+               backgroundColor: 'bg-blue-500'
+            })
+            break
+         }
+
+         default:
+            break
+      }
+   }
+
+   useEffect(() => {
```

Comment:
once you are done with the previous changes we can consider about create a custom hook that handles the alert component logic